src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/SwitchManagerTopology.java

```diff
@@ -85,22 +94,34 @@ public StormTopology createTopology() {
                 .workerSpoutComponent(WORKER_SPOUT)
                 .defaultTimeout((int) TimeUnit.SECONDS.toMillis(topologyConfig.getOperationTimeout()))
                 .build();
-        declareKafkaSpout(builder, topologyConfig.getKafkaSwitchManagerTopic(), WORKER_SPOUT);
+        declareKafkaSpout(builder, topologyConfig.getKafkaSwitchManagerTopic(), WORKER_SPOUT,
+                getZkTopoName(), getConfig().getBlueGreenMode());
         declareBolt(builder, new SpeakerWorkerBolt(speakerWorkerConfig),
                 SpeakerWorkerBolt.ID)
                 .fieldsGrouping(WORKER_SPOUT, FIELDS_KEY)
                 .fieldsGrouping(SwitchManagerHub.ID, SpeakerWorkerBolt.INCOME_STREAM, FIELDS_KEY)
                 .directGrouping(CoordinatorBolt.ID);
 
-        declareBolt(builder, buildKafkaBolt(topologyConfig.getKafkaNorthboundTopic()), NB_KAFKA_BOLT)
+        declareBolt(builder, buildKafkaBolt(topologyConfig.getKafkaNorthboundTopic(),
+                getZkTopoName(), getConfig().getBlueGreenMode()), NB_KAFKA_BOLT)
```

Comment:
Does it make sense to introduce a dedicated "build" method instead of repeating "getZkTopoName(), getConfig().getBlueGreenMode()" each time?